src/fileSystem/index.ts

```diff
@@ -54,6 +56,55 @@ export class GistFileSystemProvider implements FileSystemProvider {
     ._onDidChangeFile.event;
 
   constructor(private store: Store) {
+
+    window.tabGroups.onDidChangeTabs(async (tabGroups) => {
+      if (tabGroups.closed.length === 0) {
+        return;
+      }
+
+      for (const tab of tabGroups.closed) {
+        const input = tab.input as TabInputText;
+        if (!(input instanceof TabInputText) || !input.uri) {
+          continue;
+        }
+
+        const uri = input.uri.toString();
+        if (!uri.startsWith(FS_SCHEME) || !this.store.unsyncedFiles.has(uri)) {
+          continue; 
+        }
+
+        const filename = path.basename(Uri.parse(uri).path);
+        const { gistId } = getGistDetailsFromUri(Uri.parse(uri));
+
+        const result = await window.showWarningMessage(
+          `${filename} has changes that have not been synced to GitHub. Do you want to sync them now?`,
+          { modal: true },
+          "Yes", 
```

Comment:
I generally prefer to avoid yes/no buttons in dialogs, as opposed to making the buttons state the action. So let's call this `Sync changes`.